### DISCUSSION TITLE
Controller Status (DRO) display colours

### DISCUSSION THREAD

**Discussion by howardlowndes** [winder/Universal-G-Code-Sender#2368] [Help]
Is it possible to change the colours of the subject display. I have tried all of the skins and none of them change the colours from dark aqua/dark red on dark grey background, which are very hard to read.

**Comment by breiler** [comment]
No it is not possible at the moment. Any suggestions on a color theme?

**Comment by howardlowndes** [comment]
I'll point you to this site concerning screen contrasts, but with the
caveat that the selected colours need to be checked for colour blindness
(12% of males, much lower for females)

https://www.perkins.org/resource/choosing-high-contrast-color-schemes-for-low-vision/

**Reply by breiler** [reply]
In web development, WCAG (Web Content Accessibility Guidelines) is usually used for these types of decisions which among other things describes the minimum contrast levels for foreground and background colors.

And as you said, the dark aqua (#41818F) and red (#F00000) agains the dark grey (#2A2C2D) scores pretty bad according to those guidelines. I have experimented with adjusting the light levels for some colors to meet the minimum requirement for the contrast levels to start with. The color blindness is another thing that will be tricky to address as we also have some requests to support colors for laser protection glasses.


I made the darker aqua color (which is used when the component is inactive) a bit lighter to meet the minumum contrast requirement, from #41818F:
![Screenshot from 2023-11-18 20-17-46](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/c5cadeaf-24eb-45ed-bfbe-4e4265ef2c35)

To #4F9EB0:
![Screenshot from 2023-11-18 20-19-33](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/7336cd47-49da-40b7-a7da-2827771ee8ae)


I have also tried replacing the red color that is indicating that an axis is changing with the orange color, as red on black scored particularly bad:
![Screenshot from 2023-11-18 20-21-51](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/f594598f-1371-4e75-9bd5-0a237a8cc146)

It may even make sense to make it #00C900?
![Screenshot from 2023-11-18 20-47-01](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/cfc57eb4-cf71-435f-ab55-88bb5942c4b3)

**Comment by howardlowndes** [comment]
Interesting. Certainly the orange presents well on the dark grey but the
green may get lost with green glasses.
Have you tried shades of grey with a light background, say #EEE, with black
as the active #000, and a mid grey as the inactive?

**Reply by breiler** [reply]
With the three small changes so far:
* Made the dark aqua a bit brighter
* Changed the color of the coordinates that changes from red to green
* Changed the dark text in alarm to white

The color theme might not be the best, and until we build support for being able to change the color theme these small changes might be enough? 

![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/9d57c284-42b1-4a22-a0d0-b9dad6b5f0ec)

I made a filter that corresponds pretty well what I see with green laser goggles:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/527b39c8-ad60-4bb6-ad17-0b6fbae6d899)

Color deficient Protanopioa (insensetivity to red):
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/5d0953fb-bf25-410e-8672-3ab087e8970a)

Color deficient Deuteranopia (insensetivity to green):
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/fa07f7ef-0764-4fac-b707-dec507f5bce1)


Color deficient Tritanopia (insensetivity to blue):
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/ba003491-b49c-43e4-9f2b-76def65713a3)

**Comment by howardlowndes** [comment]
Well done, they look distinct improvements from many aspects!
Any ideas on implementation times?
I'm happy I have contributed to this discussion.

**Reply by breiler** [reply]
Thanks for the feedback. 

It is fixed here #2369 and is available in the latest nightly build.
It will available in the next version 2.1.3.
